projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs

```diff
@@ -0,0 +1,232 @@
+        internal const string MessageConversationId = "messaging.message.conversation_id";
+        internal const string MessagingOperation = "messaging.operation";
+        internal const string MessagingSystem = "messaging.system";
+        internal const string MessagingDestination = "messaging.destination.name";
+        internal const string MessagingDestinationRoutingKey = "messaging.rabbitmq.destination.routing_key";
+        internal const string MessagingBodySize = "messaging.message.body.size";
+        internal const string MessagingEnvelopeSize = "messaging.message.envelope.size";
+        internal const string ProtocolName = "network.protocol.name";
+        internal const string ProtocolVersion = "network.protocol.version";
+        internal const string RabbitMQDeliveryTag = "messaging.rabbitmq.delivery_tag";
+
+        internal static readonly ActivitySource s_source = new ActivitySource(Name,
+            typeof(RabbitMQActivitySource).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                ?.InformationalVersion ?? "");
+
+        public const string Name = "RabbitMQ.Client";
+        public static bool UseRoutingKeyAsOperationName { get; set; } = true;
+        internal static bool HasListeners => s_source.HasListeners();
+
+        internal static readonly IEnumerable<KeyValuePair<string, object>> CreationTags = new[]
+        {
+            new KeyValuePair<string, object>(MessagingSystem, "rabbitmq"),
+            new KeyValuePair<string, object>(ProtocolName, "amqp"),
+            new KeyValuePair<string, object>(ProtocolVersion, "0.9.1")
+        };
+
+        internal static Activity Send(string routingKey, string exchange, int bodySize,
+            ActivityContext linkedContext = default)
+        {
+            if (s_source.HasListeners())
+            {
+                Activity activity = linkedContext == default
+                    ? StartRabbitMQActivity(UseRoutingKeyAsOperationName ? $"{routingKey} publish" : "publish",
+                        ActivityKind.Producer)
+                    : StartLinkedRabbitMQActivity(UseRoutingKeyAsOperationName ? $"{routingKey} publish" : "publish",
+                        ActivityKind.Producer, linkedContext, Activity.Current?.Context ?? default);
+                if (activity?.IsAllDataRequested == true)
+                {
+                    PopulateMessagingTags("publish", routingKey, exchange, 0, bodySize, activity);
+                }
+
+                return activity;
+            }
+
+            return null;
+        }
+
+        internal static Activity ReceiveEmpty(string queue)
+        {
+            if (!s_source.HasListeners())
+            {
+                return null;
+            }
+
+            Activity activity = StartRabbitMQActivity(UseRoutingKeyAsOperationName ? $"{queue} receive" : "receive",
+                ActivityKind.Consumer, Activity.Current?.Context ?? default);
```

Comment:
nit: here and in other places: `Activity.Current?.Context ?? default` should not be necessary.  `default` parent context implies using `Activity.Current` as a parent